assets/add-to-cart.js

```diff
@@ -31,17 +24,208 @@ async function addToCart(snippetId) {
     if (response.error) throw new Error(response.error);
 
     const cart = document.querySelector('#cart-items-badge');
+    const cartDrawer = document.querySelector('.cart-drawer');
 
     if (cart) {
-      let cartBadgeBudge = Number(cart.innerHTML);
-
-      cart.innerHTML = ++cartBadgeBudge;
+      let cartBadgeBudge = response.count;
+      cart.innerHTML = cartBadgeBudge;
+      
+      // Update the cart drawer
+      if (cartDrawer) {
+        await updateCartDrawer();
+      }
     }
 
     stopLoad('#loading__cart');
     notify(ADD_TO_CART_EXPECTED_ERRORS.product_added, 'success');
+    toggleCartDrawer();
   } catch (err) {
     stopLoad('#loading__cart');
     notify(err.message, 'error');
   }
 }
+
```

Comment:
I don't believe it is necessary to add it because you can determine its purpose from the function's name alone.

Reply:
if you look at product.js we are using this method because the function name doesn't explain everything also that can be easy for everyone to check easily the code without spending time to define what is the code for